src/main/scala/org/broadinstitute/dsde/snoop/ws/SnoopApiService.scala

```diff
@@ -17,21 +23,54 @@ object SnoopApiServiceActor {
   }
 }
 
-class SnoopApiServiceActor(override val executionServiceHandler: RequestContext => WorkflowExecutionService) extends Actor with SnoopApiService {
+class SnoopApiServiceActor(override val executionServiceHandler: RequestContext => WorkflowExecutionService) extends Actor with BaseSnoopApiService with WorkflowExecutionApiService {
+  implicit def executionContext = actorRefFactory.dispatcher
   def actorRefFactory = context
-  def receive = runRoute(snoopRoute)
-}
+  def possibleRoutes = baseRoute ~ workflowRoutes ~ swaggerService.routes 
 
+  def receive = runRoute(possibleRoutes)
 
-// this trait defines our service behavior independently from the service actor
-trait SnoopApiService extends HttpService {
-  implicit def executionContext = actorRefFactory.dispatcher
+  private val snoopConfig: Config = ConfigFactory.parseFile(new File("/etc/snoop.conf"))
+  private val swaggerConfig = snoopConfig.getConfig("swagger")
+
+  val swaggerService = new SwaggerHttpService {
+    override def apiTypes = Seq(typeOf[BaseSnoopApiService], typeOf[WorkflowExecutionApiService])
+
+    override def apiVersion = swaggerConfig.getString("apiVersion")
+    override def baseUrl = swaggerConfig.getString("baseUrl")
+    override def docsPath = swaggerConfig.getString("apiDocs")
+    override def actorRefFactory = context
+    override def swaggerVersion = swaggerConfig.getString("swaggerVersion")
+
+    override def apiInfo = Some(
+      new ApiInfo(
+        swaggerConfig.getString("info"),
+        swaggerConfig.getString("description"),
+        swaggerConfig.getString("termsOfServiceUrl"),
+        swaggerConfig.getString("contact"),
+        swaggerConfig.getString("license"),
+        swaggerConfig.getString("licenseUrl"))
+    )
+  }
+}
+
+@Api(value = "", description = "Snoop Base API", position = 1)
+trait BaseSnoopApiService extends HttpService {
```

Comment:
I would call this Root, not Base. The latter makes me think that something extends this.